Fix invalid host header errors (#4684)
backstage/backstage: contrib/docs/tutorials/help-im-behind-a-corporate-proxy.md

```

4. Start the backend with `yarn start`

## config

> If your development environnement is in the cloud (like with [AWS Cloud9](https://aws.amazon.com/cloud9/) or an instance of [Theia](https://theia-ide.org/)), you will need to update your configuration.

Reviewer: ```suggestion
If your development environment is in the cloud (like with [AWS Cloud9](https://aws.amazon.com/cloud9/) or an instance of [Theia](https://theia-ide.org/)), you will need to update your configuration.
```

> In `app.local.config`, you will probably need to make some changes, the exact values will depend of your setup.

Reviewer: ```suggestion
In `app.local.config`, you will probably need to make some changes, the exact values will depend on your setup.
```

> In `app.local.config`, you will probably need to make some changes, the exact values will depend of your setup.

Reviewer 2: ```suggestion
In `app-config.local.yaml`, you will probably need to make some changes, the exact values will depend on your setup.
```

That's the config you are referring to right?

Author: Of course yes

> In `app-config.local.yaml`, you will probably need to make some changes, the exact values will depend on your setup.

Reviewer 3: Should just make this `app-config.yaml` imo, or perhaps say there or in a custom config file. The `.local.yaml` one is not committed, so not a great place to put config you care about

Author: That's right, personally I fell this sort of configuration is better in an local, git-ignored file but I understand this might be confusing, I've rephrased that.

For instance, if your public url is `https://your-public-url.com` and the port `3000` and `8080` are proxified as is:

```yaml
app:
  baseUrl: https://your-public-url.com:3000
  listen:
    host: 0.0.0.0 # This make the dev server bind to localhost and not the baseUrl hostname

backend:
  baseUrl: https://your-public-url.com:8080
  listen:
    port: 8080
  cors:
    origin: https://your-public-url.com:3000
```

> If the protocole is `http`, you will need to set `backend.csp.upgrade-insecure-requests` to `false` as well.

Reviewer: ```suggestion
If the protocol is `http`, you will need to set `backend.csp.upgrade-insecure-requests` to `false` as well.
```

> If the protocole is `http`, you will need to set `backend.csp.upgrade-insecure-requests` to `false` as well.

Reviewer 2: ```suggestion
If the protocol is `http`, you will need to set `backend.csp.upgrade-insecure-requests` to `false` as well.
```

> The app port must proxy websocket connection in order to have hot reloading.

Reviewer 2: ```suggestion
The app port must proxy websocket connections in order to make hot reloading work.
```
